src/Underworld.ts

```diff
@@ -3472,7 +3473,7 @@ ${CardUI.cardListToImages(player.stats.longestSpell)}
     }
     switch (u.unitSubType) {
       case UnitSubType.MELEE:
-        const maxPathDistance = u.attackRange + u.stamina;
+        const maxPathDistance = u.attackRange + (Unit.canMove(u) ? u.stamina : 0);
```

Comment:
Just curious, was there a found bug that this fixes or did you just notice it was missing?  Just wondering under what circumstances this would matter, I don't recall prediction not working for frozen units for example

Reply:
This fixed a prediction issue where a melee unit with dark tide would still show as attacking the player.

Freeze didn't have the same issue is because frozen units **can't act** at all and are skipped in the smart targeting function before this logic runs. Dark Tide units **can act** (i.e. archers shooting from the liquid), they just can't move, and whether or not they could move wasn't being checked here, only their stamina.